File: lib/java/opentoken/src/main/java/com/truveta/opentoken/attributes/validation/NotNullOrEmptyValidator.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.attributes.validation;

/**
 * A Validator that asserts the value is <b>NOT</b> <code>null</code>
 * and blank.
 */
public final class NotNullOrEmptyValidator implements SerializableAttributeValidator {

    private static final long serialVersionUID = 1L;

    /**
     * Validates that the attribute value is not <code>null</code> or blank.
     */
    @Override
    public boolean eval(String value) {
        return value != null && !value.isBlank();
    }

}

File: lib/python/opentoken/src/main/opentoken/attributes/validation/not_null_or_empty_validator.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from opentoken.attributes.validation.serializable_attribute_validator import SerializableAttributeValidator


class NotNullOrEmptyValidator(SerializableAttributeValidator):
    """
    A Validator that asserts the value is NOT None and not blank.
    """

    def eval(self, value: str) -> bool:
        """
        Validate that the attribute value is not None or blank.

        Args:
            value: The attribute value to validate

        Returns:
            True if value is not None and not blank, False otherwise
        """
        return value is not None and value.strip() != ""
